src/test/java/Homework23.java

```diff
@@ -0,0 +1,37 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class Homework23 extends BaseTest {
+
+    public void login() {
+
+        provideEmail("tesfaye.abagaz@testpro.io");
+        providePassword("te$t$tudent");
+        clickSubmit();
+
+    }
+
+    public void provideEmail(String email) {
+        WebElement emailField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='email']")));
+        //WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        emailField.click();
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public void providePassword(String password) {
+        WebElement passwordField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='password']")));
+        //WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        passwordField.click();
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public void clickSubmit() {
+        WebElement submit = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("button[type='submit']")));
+        //WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        submit.click();
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.


![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/99f587d8-9945-46ed-abac-36bb0ab1554d)

For example

![Clean_Test](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/5bd571c5-be99-433c-965f-8dfbfdc09d27)